### PR TITLE
Add 4th pickle protocol version to tests

### PR DESCRIPTION
**Pull request by gyermolenko**
also fix deprecation warning with raw string in one test

### CODE REVIEW

File: tests/gen_pickles.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def write(name, proto):
 def generate():
     if not USE_CYTHON:
         raise RuntimeError("Cython is required")
-    for proto in range(pickle.HIGHEST_PROTOCOL):
+    for proto in range(pickle.HIGHEST_PROTOCOL+1):
```
Comment (webknjaz) [inline tests/gen_pickles.py]: please use spaces around operators.

Comment (gyermolenko) [reply tests/gen_pickles.py]: flake8 was alright with that version.
I don't mind the change but do you have any refs to support spacing?

Comment (webknjaz) [reply tests/gen_pickles.py]: It's more readable and a common standard to put spaces around operator (except in kwargs in function calls)

Comment (webknjaz) [reply tests/gen_pickles.py]: For example, https://www.python.org/dev/peps/pep-0008/#id28

Comment (Congee) [reply tests/gen_pickles.py]: Any reason it's not merged yet?

Comment (webknjaz) [reply tests/gen_pickles.py]: Just that we're all busy humans :)

**Pull request merged**